### PR TITLE
Add support for golang 1.21

### PR DESCRIPTION
**Pull request by krishnamiriyala**

### CODE REVIEW

File: pkg/datastore/database.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ func (db *relationalDb) getDBTransaction(ctx context.Context, tableName string,
 			TABLE_NAME:   tableName,
 			"authorizer": TypeName(db.authorizer),
 		})
-		db.logger.Error(err)
+		db.logger.Debug(err)
```
Comment (jeyhun) [inline pkg/datastore/database.go:115]: Why change logging level to Debug? Isn't this an actual error that needs to be logged at Error level?

File: pkg/errors/error_codes.go [hunk 2]
```diff
@@ -45,8 +45,12 @@ const (
 )
 
 func contextToString(ctx context.Context) string {
-	contextType := reflect.TypeOf(ctx).Elem()
-	if contextType.Kind() != reflect.Struct {
+	typ := reflect.TypeOf(ctx)
+	if fmt.Sprintf("XXX%s", typ) == "XXXcontext.backgroundCtx" {
```
Comment (jeyhun) [inline pkg/errors/error_codes.go:49]: just wondering, what's the point of prepending "XXX" here? Why not do `if fmt.Sprintf("%s", typ) == "context.backgroundCtx" {}`?

File: pkg/datastore/database.go (hunk 1 above)
Comment (krishnamiriyala) [reply pkg/datastore/database.go:115]: Never log the bare error without context, it's useless, consumer can log the error if its a failure case.

File: pkg/errors/error_codes.go (hunk 2 above)
Comment (krishnamiriyala) [reply pkg/errors/error_codes.go:49]: makes it easier when logging if the string is empty.

**Review by jeyhun — APPROVED** [review]

**Pull request merged**